db/model/Gdoc/archieToEnriched.ts

```diff
@@ -62,89 +141,40 @@ export const archieToEnriched = (text: string): OwidGdocContent => {
     const parsed: any = lowercaseObjectKeys(parsed_unsanitized)
 
     const toc: TocHeadingWithTitleSupertitle[] = []
-    let pointer: Array<string | number> = []
-    // archie doesn't have a nested list structure. it treats as a series of text blocks
-    // we want to put them into a nested (for now only <ul>) structure
-    // we create a copy of where the list began so that we can push its siblings into it
-    let listPointer: Array<string | number> = []
-    let isInList = false
 
     // Traverse the tree, tracking a pointer and nesting when appropriate
     function traverseBlocks(
-        value: OwidRawGdocBlock,
-        callback: (child: OwidRawGdocBlock) => void
+        node: OwidRawGdocBlock | OwidRawGdocBlock[],
+        callback: (node: OwidRawGdocBlock) => void
     ): void {
-        // top-level
-        if (isArray(value)) {
-            value.forEach((value, index) => {
-                pointer[0] = index
-                traverseBlocks(value, callback)
-            })
-        } else if (value.type === "gray-section") {
-            const pointerLength = pointer.length
-            value.value.forEach((value, index) => {
-                pointer[pointerLength] = index
-                traverseBlocks(value, callback)
-            })
-            pointer = pointer.slice(0, -1)
+        if (isArray(node)) {
+            node.forEach((value) => traverseBlocks(value, callback))
+        } else if (node.type === "gray-section") {
+            traverseBlocks(node.value, callback)
```

Comment:
This is much nicer than before! I have a bit of the same gut feeling as with the code in utils/Util.ts - I think it would be nice if we could extract tree traversal from node handling so it that you just have to specify once how to traverse a tree and somewhere else if your components has links to track and you don't have to search the codebase for places to adjust this.

Maybe best not to hold up this PR with this but maybe let's chat at some point about the options in that space.

Reply:
If we do this, it'd also be nice to do this toc thing after we've enriched the tree.

 Having two _almost-but-not-quite_ similar functions because one's Raw and the other Enriched is also annoying.